rostime/include/ros/impl/duration.h

```diff
@@ -56,7 +56,7 @@ namespace ros {
     if (sec64 < INT_MIN || sec64 > INT_MAX)
       throw std::runtime_error("Duration is out of dual 32-bit range");
     sec = (int32_t)sec64;
-    nsec = (int32_t)((d - (double)sec)*1000000000);
+    nsec = (int32_t)((d - (double)sec)*10000000000/10);
```

Comment:
The approach to multiply it with an order of magnitude more and then divide again doesn't looks like a good approach.

Can you please try to use a rounding function (e.g. `nearbyint`) instead and confirm that it still addressed your problem.

Reply:
yes, `nearbyint` does the job.

Reply:
I updated the PR accordingly